src/parser.js

```diff
@@ -221,7 +218,31 @@ define([
         mug.p.rawDataAttributes = getAttributes(el);
         return mug;
     }
-            
+
+    function parseSetValues(form, setValues) {
+        var rootNodeName = form.tree.getRootNode().getID();
+
+        setValues.each(function () {
+            var $el = $(this);
+            form.vellum.parseSetValue(
+                form, $el, processPath($el.attr('ref'), rootNodeName));
+        });
+    }
+
+    function parseSetValue(form, el, path) {
+        var mug = form.getMugByPath(path),
+            event = el.attr('event'),
+            ref = el.attr('ref'),
+            value = el.attr('value');
+
+        // HACK: hardcoding these as that's what setValue will support for now
+        if (!mug || (event !== 'xforms-ready' && event !== 'jr-insert')) {
+            form.addSetValue(event, ref, value);
+        } else {
+            mug.p.setValue = value;
```

Comment:
For some mug types setValue is presence is notallowed. This could cause a validation error that the user cannot fix. Instead of setting the spec to `{presence: 'notallowed'}`, I suggest setting it to `{presence: 'optional', visibility: 'hidden'}`. I'm 70% confident that should not cause problems elsewhere...